source/Engine.cpp

```diff
@@ -1578,6 +1582,42 @@ void Engine::SendHails()
 
 
 
+// Handle any keyboard inputs for the engine. This is done in the main thread
+// after all calculation threads are paused to avoid race conditions.
+void Engine::HandleKeyboardInputs()
+{
+	Ship *flagship = player.Flagship();
+	
+	// Commands can't be issued if your flagship is dead.
+	if(!flagship || flagship->IsDestroyed())
+		return;
+	
+	// Determine which new keys we press
+	Command oldHeld = keyHeld;
+	keyHeld.ReadKeyboard();
+	Command keyDown = keyHeld.AndNot(oldHeld);
+	
+	++landKeyInterval;
+	if(oldHeld.Has(Command::LAND))
+		landKeyInterval = 0;
+	
+	// Wait with actual jumping until the jump command is released
+	// Or if pressing land quicky in succession, then use WAIT to switch landing targets
```

Comment:
Perhaps we should document the interpretations of Command::WAIT in Command.h/.cpp

Reply:
Good idea. I added this documentation.